tests/test_context.py

```diff
@@ -48,3 +52,49 @@ def test_message_handler(device):
     assert context.message_handler is message_handler_test
     with context:
         assert current_context().message_handler is context.message_handler
+
+
+def test_async_wake_interval(device):
+    """Test read-write property async_wake_interval."""
+    with Context(device) as context:
+        context.async_wake_interval = 42
+        assert context.async_wake_interval == 42
+
+
+def test_default_resampler_index(device):
+    """Test return values default_resampler_index"""
+    with Context(device) as context:
+        assert context.default_resampler_index >= 0
+        index = context.default_resampler_index
```

Comment:
Could you please move this up so `context.default_resampler_index` doesn't needs to be evaluated twice?  Also extreme nit: docstring is missing the period.

Reply:
nice catch!